src/mimefactory.rs

```diff
@@ -659,6 +615,92 @@ impl<'a> MimeFactory<'a> {
+                "hidden-recipients".to_string(),
+                Vec::new(),
+            ));
+        } else {
+            let email_to_remove =
+                if self.msg.param.get_cmd() == SystemMessage::MemberRemovedFromGroup {
+                    self.msg.param.get(Param::Arg)
+                } else {
+                    None
+                };
+
+            for (name, addr) in &self.recipients {
+                if let Some(email_to_remove) = email_to_remove {
+                    if email_to_remove == addr {
+                        continue;
+                    }
+                }
+
+                if name.is_empty() {
+                    to.push(Address::new_mailbox(addr.clone()));
+                } else {
+                    to.push(Address::new_mailbox_with_name(
+                        name.to_string(),
+                        addr.clone(),
+                    ));
+                }
+            }
+
+            if to.is_empty() {
+                to.push(from.clone());
+            }
+        }
+
+        // Start with Internet Message Format headers in the order of the standard example
+        // <https://datatracker.ietf.org/doc/html/rfc5322#appendix-A.1.1>.
+        let from_header = Header::new_with_value("From".into(), vec![from]).unwrap();
+
+        if is_encrypted {
+            warn!(context, "Sending encrypted message with display names etc.");
+
+            // Add receipients and sender with display names to the protected headers
+            headers
+                .protected
+                .push(Header::new_with_value("To".into(), to.clone()).unwrap());
+
+            headers.protected.push(from_header.clone());
+
+            // Add receipients and sender without display names to the unprotected headers
+            headers.unprotected.push(
+                Header::new_with_value(
+                    "To".into(),
+                    to.into_iter()
+                        .map(|header| match header {
+                            Address::Mailbox(mb) => Address::new_mailbox(mb.address),
+                            Address::Group(name, participants) => Address::new_group(
+                                name,
+                                participants
+                                    .into_iter()
+                                    .map(|mb| Mailbox::new(mb.address))
+                                    .collect(),
+                            ),
+                        })
+                        .collect::<Vec<_>>(),
+                )
+                .unwrap(),
+            );
+
+            headers.unprotected.push(
+                Header::new_with_value(
+                    "From".into(),
+                    vec![Address::new_mailbox(self.from_addr.clone())],
+                )
+                .unwrap(),
+            );
+        } else {
+            // Unencrypted emails have display names in the unprotected headers
+            headers
```

Comment:
Can be put to `protected` to merge the code in both branches. Anyway for unencrypted messages `protected` headers are sent out too

Reply:
I don't know what you mean by that

Reply:
In the else case https://github.com/deltachat/deltachat-core-rust/blob/d3dbc3f2245e3090f26ee41fe675dd612d77534c/src/mimefactory.rs#L861-L862, the headers will then go into the child-part, but that is fine I guess.